src/Database/Oracle/Simple/ToField.hs

```diff
@@ -27,3 +30,25 @@ instance ToField Int where
 
 instance ToField DPITimestamp where
   toField = pure . AsTimestamp
+
+instance ToField UTCTime where
+  toField utcTime = pure (AsTimestamp dpiTs)
```

Comment:
This could also be exported as a "smart" constructor for DPITimestamp
```haskell
mkDPITimestamp :: UTCTime -> DPITimestamp
mkDPITimestamp utcTime = ...

instance ToField UTCTime where
  toField = pure . AsTimestamp . mkDPITimestamp
```

Reply:
That's a great idea, yea we probably shouldn't export the type at all, just keep it internal. The conversion functions can be used in `ToField` / `FromField` and then we can those functions as well for roundtrip property-based tests.